File: functions/src/types/pendingUser.ts
import {Gender} from "./gender";

export interface PendingMeasurement {
  date: number;
  height: number;
  weight: number;
  neck: number;
  biceps: number;
  chest: number;
  waist: number;
  belt: number;
  hips: number;
  thigh: number;
  calf: number;
}

export interface PendingUser {
  email: string;
  gender: Gender;
  firstAndLastName: string;
  storedAge: number;
  lastUpdated: Date;
  measurements: PendingMeasurement[];
}

File: app/src/main/java/com/noisevisionsoftware/vitema/domain/model/user/pending/PendingUser.kt
package com.noisevisionsoftware.vitema.domain.model.user.pending

import com.noisevisionsoftware.vitema.domain.model.user.Gender
import java.util.Date

data class PendingUser (
    val email: String = "",
    val gender: Gender = Gender.OTHER,
    val age: Int = 0,
    val firstAndLastName: String = "",
    val lastUpdated: Date = Date(),
    val measurements: List<PendingMeasurement>? = null
)